bfasst/status.py

```diff
@@ -97,18 +97,25 @@ class MapStatus(enum.Enum):
 
 
 class BfasstException(Exception):
-    def __init__(self, err, msg):
+    def __init__(self, err, msg, creator=None):
         super().__init__(msg)
         self.error = err
+        self.msg = msg
+        self.creator = creator
 
 
 class Status:
+    """Represents the status of a BFASST tool after it has run"""
     def __init__(self, status, msg="", raise_excep=True):
         self.status = status
         self.msg = msg if msg else ""
         self.error = bool(status.value)
         if status.value and raise_excep:
-            raise BfasstException(status, f": {self.msg}")
+            if msg and msg not in msg_map[self.status]:
+                raise BfasstException(status, f"{msg_map[self.status]}: {self.msg}", self)
+            raise BfasstException(status, msg_map[self.status], self)
 
     def __str__(self):
-        return f"{msg_map[self.status]}{self.msg}"
+        if self.msg not in msg_map[self.status]:
+            return f"{msg_map[self.status]}: {self.msg}"
+        return msg_map[self.status]
```

Comment:
Can you add some comments here explaining why this approach was necessary?  Thanks!

Reply:
Done